sonar-plugin-api/src/main/java/org/sonar/api/server/ws/LocalConnector.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.ws;
+
+import com.google.common.annotations.Beta;
+import java.util.Collection;
+import javax.annotation.CheckForNull;
+
+/**
+ * This class allows a web service to call another web service through the sonar-ws library.
+ * The call is in-process, synchronous and does not involve the HTTP stack.
+ * <p>
+ * Example of a web service that uses sonar-ws 5.5 to get some issues:
+ * <pre>
+ * public class MyRequestHandler implements org.sonar.api.server.ws.RequestHandler {
+ *   {@literal @}Override
+ *   public void handle(Request request, Response response) {
+ *     WsClient wsClient = WsClientFactory.newLocalClient(request.getLocalConnector());
+ *     SearchWsResponse issues = wsClient.issues().search(new SearchWsRequest());
+ *     // ...
+ *   }
+ * }
+ * </pre>
+ * </p>
+ * @since 5.5
+ */
+@Beta
+public interface LocalConnector {
+
+  interface LocalRequest {
+    /**
+     * URL path, which is the concatenation of controller path and action key, for example "api/issues/search"
+     * @see org.sonar.api.server.ws.WebService.Controller#path
+     * @see org.sonar.api.server.ws.WebService.Action#key
+     */
+    String getPath();
+
+    /**
+     * @see Request#getMediaType()
+     */
+    String getMediaType();
+
+    /**
+     * HTTP method. Possible values are "GET" and "POST"
+     * @see Request#method()
+     */
+    String getMethod();
```

Comment:
shouldn't there be constants in the API representing the values `GET` and `POST` to avoid user define there own constants (or use magic strings in there code)?